src/linear_algebra.jl

```diff
@@ -0,0 +1,91 @@
+for (jf, af) in [(:+, :add!), (:-, :sub!)]
+    @eval function Base.$jf(A::T, B::T) where {T<:Union{ArbMatrix,AcbMatrix}}
+        @boundscheck (
+            size(A) == size(B) ||
+            throw(DimensionMismatch("Matrix sizes are not compatible."))
+        )
```

Comment:
I don't think it's common to use `@boundscheck` here, from what I can tell `Base` doesn't do it.

Reply:
Similarly for the other cases where this is done.

Reply:
Otherwise the check looks good!

Reply:
Well, this just means that an @inline will elide this check. Not that it matters much at all.

Reply:
I also think this is used quite a lot in Base, e.g., https://github.com/JuliaLang/julia/blob/cebd4fa959b6f9835a3abe70e705136b5e914e4a/stdlib/LinearAlgebra/src/bitarray.jl#L98

Reply:
Okay Base seems to vary a bit, lets keep it then!